master/buildbot/test/unit/test_interpolate_secrets.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import absolute_import
+from __future__ import print_function
+
+from twisted.internet import defer
+from twisted.trial import unittest
+
+from buildbot.process.properties import Interpolate
+from buildbot.secrets.providers.base import SecretProviderBase
+from buildbot.secrets.secret import SecretDetails
+from buildbot.test.fake import fakemaster
+from buildbot.test.fake.fakebuild import FakeBuild
+from buildbot.test.util.config import ConfigErrorsMixin
+
+
+class FakeSecretStorage(SecretProviderBase):
```

Comment:
Didn't I see this fake already? I think you should put it in buildbot.test.fake.secrets